go/vrf/conversion.go

```diff
@@ -79,34 +78,30 @@ func secg1EncodeCompressed(curve elliptic.Curve, x, y *big.Int) []byte {
 
 var errInvalidPoint = errors.New("invalid point")
 
-func secg1Decode(curve elliptic.Curve, data []byte) (x, y *big.Int, err error) {
+func unmarshalCompressed(curve elliptic.Curve, data []byte) (x, y *big.Int, err error) {
 	byteLen := (curve.Params().BitSize + 7) >> 3
-	if (data[0] &^ 1) != 2 {
+	if (data[0] &^ 1) != 2 { // compressed form
 		return nil, nil, errors.New("unrecognized point encoding")
 	}
 	if len(data) != 1+byteLen {
 		return nil, nil, errors.New("invalid length for curve")
 	}
 
 	// Based on Routine 2.2.4 in NIST Mathematical routines paper
-	params := curve.Params()
-	tx := new(big.Int).SetBytes(data[1 : 1+byteLen])
-	y2 := y2(params, tx)
-	sqrt := defaultSqrt
-	ty := sqrt(y2, params.P)
-	if ty == nil {
+	x = new(big.Int).SetBytes(data[1:])
+	y2 := y2(curve.Params(), x)
+	y = new(big.Int).ModSqrt(y2, curve.Params().P)
+	if y == nil {
 		return nil, nil, errInvalidPoint // "y^2" is not a square
 	}
-	var y2c big.Int
-	y2c.Mul(ty, ty).Mod(&y2c, params.P)
-	if y2c.Cmp(y2) != 0 {
+	if !curve.IsOnCurve(x, y) {
 		return nil, nil, errInvalidPoint // sqrt(y2)^2 != y2: invalid point
 	}
-	if ty.Bit(0) != uint(data[0]&1) {
-		ty.Sub(params.P, ty)
+	if y.Bit(0) != uint(data[0]&0x01) {
```

Comment:
[optional] I think it's more readable to split the operator. So:
data[0] & 0x01 vs data[0]&0x01
If there are overriding style suggestions that cut against this, feel free to ignore.

Reply:
`gofmt` removes the space unfortunately.